### PR TITLE
certificate credit systems

### LINKED ISSUE #468: Finalising certification
Basically we need a few more views to close the loop:

1) A view for an **organisation** owner which lists all certificates which have been issued but not paid for in their organisation (with a total after indicating the cost of outstaninding invoices)
Course type, Course date, User name & surname, Certificate link

http://staging.changelog.qgis.org/en/test/unpaidcertificates/list/

2) A view for a **project manager** which lists same as above but grouped by organisation

http://staging.changelog.qgis.org/en/test/unpaidcertificates/list/

```
QGIS Unpaid Certificates

Organisation: Foo

Course type, Course date, User name & surname, Certificate link
Course type, Course date, User name & surname, Certificate link
Course type, Course date, User name & surname, Certificate link
Course type, Course date, User name & surname, Certificate link
...
Course type, Course date, User name & surname, Certificate link

Total outstanding: USD 400

Org
```

3) A form for the **project manager** to register that he has recieved payment from an organisation owner. When he accepts the form, the received total is divided by the certificate unit price (to produce X) and then X items in the list of items unpaid for in the organisation are removed from the unpaid list (oldest first). A cross reference in the certificate record should be made to the payement record so you can see when the certificate was paid for

```
                        ___________                   
Organisation: |   Foo | v |
                        -----------

Amount outstanding: $400

Record payment: 

OK
```

4) a form that a project administrator can use to set the price for certificates
if the price is set to zero then the  view above  in organization is hidden (i.e. we hide the list of unpaid  certificates)

5) When recording payments for a batch of certificates, the unit prices should also be shown  so that if the project admin changes the price at the provject level, it is clear what the individual certificates costed

The revised schema (new entities in blue) might look like this:

screen shot 2017-07-12 at 4 01 13 pm

See @cchristelis  alternative approach below.

### PR DESCRIPTION
**Pull request by ann26**
fix #468 

This PR:
- [x] modify project model to include certificate cost
- [x] modify organisation model to include certificate credits
- [x] every time a certificate is issued, the certificate credits of the organisation will be deducted by 1

It still allows certificate credits to be minus.

### CODE REVIEW

File: django_project/certification/views/certificate.py [hunk 1]
```diff
@@ -97,6 +97,15 @@ def form_valid(self, form):
 
         try:
             super(CertificateCreateView, self).form_valid(form)
+
+            # Update organisation credits every time a certificate is issued.
+            organisation = \
+                CertifyingOrganisation.objects.get(
+                    slug=self.organisation_slug)
+            remaining_credits = organisation.credits_certificate - 1
```
Comment (cchristelis) [inline django_project/certification/views/certificate.py]: This one cannot be hard-coded. Let's put it in the project settings model.

**Comment by ann26** [comment]
@timlinux and @cchristelis 
here is update:
In here, the example is a certificate cost 5 credit (editable in project model).
When the credits is insufficient, icons changed to `top up icons`
peek 2017-09-08 18-33

when credits is topped up, user may choose which certificate they want to pay
peek 2017-09-08 18-50

`top up` icons and `pay with credits` icons is not working yet

**Comment by ann26** [comment]
Pay unpaid certificates using credits that has just been top up
peek 2017-09-11 12-12

**Comment by ann26** [comment]
current top up view:
screenshot from 2017-09-11 13-13-41

**Comment by timlinux** [comment]
@ann26 I'm going to merge this as we have another user who wants to test the system. Please come back after and add tests ok?

**Comment by ann26** [comment]
ok @timlinux

**Pull request merged**